modules/c++/tiff/source/Common.cpp

```diff
@@ -39,22 +41,33 @@ std::string tiff::RationalPrintStrategy::toString(const sys::Uint32_T data)
     return tempStream.str();
 }
 
+template<typename T>
+static inline void memcpy_(T* pDest, const void* pSrc)
+{
+    memcpy(pDest, pSrc, sizeof(T));
+}
+
 sys::Uint64_T tiff::combine(sys::Uint32_T numerator,
         sys::Uint32_T denominator)
 {
     sys::Uint64_T value;
-    sys::Uint32_T *ptr = (sys::Uint32_T *)&value;
 
-    memcpy(ptr, &numerator, sizeof(sys::Uint32_T));
-    ptr++;
-    memcpy(ptr, &denominator, sizeof(sys::Uint32_T));
+    //sys::Uint32_T *ptr = (sys::Uint32_T *)&value;
+    auto ptr = reinterpret_cast<std::byte*>(&value);  // reinterpret_cast<> to std::byte is allowed by the standard
+    memcpy_(ptr, &numerator);
+    memcpy_(ptr + sizeof(sys::Uint32_T), &denominator);
 
     return value;
 }
 void tiff::split(sys::Uint64_T value, sys::Uint32_T &numerator,
                  sys::Uint32_T &denominator)
 {
-    numerator = ((sys::Uint32_T*)&value)[0];
-    denominator = ((sys::Uint32_T*)&value)[1];
+    // Using casts generate a warning: dereferencing type-punned pointer will break strict-aliasing rules[-Wstrict-aliasing]
+    // Do the reverse of combine() and use memcpy().
+    //numerator = ((sys::Uint32_T*)&value)[0];
+    //denominator = ((sys::Uint32_T*)&value)[1];
+    auto ptr = reinterpret_cast<const std::byte*>(&value);  // reinterpret_cast<> to std::byte is allowed by the standard
```

Comment:
I would prefer to use `sys::ubyte*` (which is just a typedef of `unsigned char*`) here, or alternatively `unsigned char*` directly if `sys::ubyte*` gives a warning. I don't think defining `std::byte` for C++ versions < 17 ourselves is technically allowed (https://en.cppreference.com/w/cpp/language/extending_std).

Also, the GSL library has some more complicated conditions for testing whether `std::byte` is available:
https://github.com/beinhaerter/GSL/blob/128683c204616fcd441f1643cdcaf33a65a0392c/include/gsl/gsl_byte#L60-L64. I think it's better to avoid until coda-oss requires C++17.

Reply:
OK ... done.